### ISSUE TITLE
CBT (Computer-Based Testing)における QTI コンテンツの位置づけと WCAG JIS の影響について

### ISSUE THREAD

**Issue by murata2makoto** [waic/wcag22#1889]
本 issue は、日本の学校教育で用いられている CBT（Computer Based Testing）におけるQTI コンテンツの位置づけと、WCAG JIS の改定がそのアクセシビリティに与える影響について、議論の前提となる事実を整理することを目的とする。

### 1. 日本の学校教育における CBT と QTI

日本の小学校・中学校教育における CBT (特に文部科学省の[MEXCBT](https://www.mext.go.jp/a_menu/shotou/zyouhou/mext_00001.html)）では、1EdTech（旧 IMS Global）が策定した QTI（Question and Test Interoperability）が、問題・テストの記述形式として用いられている。

QTI は次のような特徴を持つ。

- QTI は HTML 仕様ではない
- QTI はXMLに基づく言語であり、問題文や設問の構造と意味を記述する
- 実行時には、ブラウザ上で Web 技術（HTML / CSS / JavaScript 等）によりレンダリングされる

### 2. WCAG JIS 改定が日本の学校教育におけるCBT に与える影響

QTI コンテンツは HTML 文書ではないが、ブラウザ上で提供・操作されるウェブコンテンツであり、WCAGの適用範囲にある。すでに、日本の CBT のアクセシビリティの評価や議論の際にWCAG 2.x が参照されているのが実情である。（例：[TAO Testing によるアクセシビリティに関する解説](https://www.taotesting.com/ja/accessibility-standards-in-online-assessment/)を参照。)

小学校・中学校教育においてアクセシビリティは極めて重要であり、これを考慮しないCBT 運用は、障害者差別解消法等との関係で問題となる可能性がある。

### 3. 縦書きとルビについて

小学校・中学校教育で用いるCBTでは、縦書き・ルビは必須の機能である。制定中の WCAG JIS が 縦書きおよびルビのアクセシビリティに対処しない場合、こうしたCBTにおいて、縦書きとルビに関するアクセシビリティ基準が存在しないことになる。

QTI 独自のアクセシビリティ基準を設けてそこで縦書きおよびルビに対処するという選択肢も理論上は考えられるが、WCAG は HTML に特化した仕様ではなく、QTI にも当然適用できるはずの仕様である。

備考: EPUBの場合もQTIと同様の事情が存在する。さらに言えば、EPUBとQTIで、縦書きおよびルビに関する別仕様を作るというのは統一性がなくなり、実装コストの増加にもつながりかねない。

**Comment by MakotoUeki** [comment]
@murata2makoto 
> 制定中の WCAG JIS が 縦書きおよびルビのアクセシビリティに対処しない場合

どのような場合がこれに該当するのか、具体例をいくつか挙げていただくことは可能でしょうか？

**Comment by momdo** [comment]
参考情報として、ウェブアクセシビリティを含む情報アクセシビリティは一般に障害者差別解消法の「環境の整備」に位置づけられており、「環境の整備」は努力義務でしかなく、国内でコンテンツがWCAG等を満たしていない状態がただちに法令違反になるわけではありません。

> (4)環境の整備との関係  
> 法第5条においては，個別の場面において，個々の障害者に対して行われる合理的配慮を的確に行うための不特定多数の障害者を主な対象として行われる事前的改善措置（施設や設備のバリアフリー化，意思表示やコミュニケーションを支援するためのサービス・介助者等の人的支援，障害者による円滑な情報の取得・利用・発信のための情報アクセシビリティの向上等をいう。以下同じ。）を，環境の整備として事業者の努力義務としていること。

[文部科学省所管事業分野における障害を理由とする差別の解消の推進に関する対応指針の策定について](https://www.mext.go.jp/a_menu/shotou/tokubetu/material/mext_02599.html)より

**Comment by murata2makoto** [comment]
WCAG 等の仕様が法律から直接参照されているわけではない以上、それを満たさない状態が直ちに法令違反になるわけではない、という点はそのとおりです。一方で、アクセシブルでない CBT を学校が生徒に課すことは、条件次第では十分に違法となり得ます。

私は、WCAG JIS がそのような事態を未然に回避するための実効的な指針として機能することを願っています。WCAG JIS は、産業標準化法第1条にいう「公共の福祉の増進に寄与すること」を目的として策定される標準のはずだからです。

**Comment by MakotoUeki** [comment]
@murata2makoto 
> - それらは結果として WCAG 2.x の達成基準に違反している可能性が高い
> - しかし「縦書き CBT 問題はどう作るべきか」「どのような実装が期待されるのか」という前向きなガイドラインが存在しない

該当する達成基準がある場合は、太田さん（@bakera)さんもおっしゃっている通り、その達成基準を満たすことができる方法（Sufficient teachnique）として新規追加の提案をAGWGに行うのがよいかと思います。

WCAGの達成基準（normative)は、既存の特定の技術に依存することなく、将来の新技術を含むあらゆる技術にも適用できるように、コンテンツがどうあるべきかを定めています。そのため、書きっぷりも抽象度が高くなっています。

そこで、各達成基準を念頭に、どう作るべき、どのような実装が期待されるのかについては、具体的なテクニック（informative）として代表的なものが文書化されています。

- G（General）：あらゆる技術に適用可能
  - 例：[G94: Providing short text alternative for non-text content that serves the same purpose and presents the same information as the non-text content](https://www.w3.org/WAI/WCAG22/Techniques/general/G94) 
- H（HTML）：HTML特有のテクニック
  - 例：[H37: Using alt attributes on img elements](https://www.w3.org/WAI/WCAG22/Techniques/html/H37)
- PDF（PDF）：PDF特有のテクニック
  - 例：[PDF1: Applying text alternatives to images with the Alt entry in PDF documents](https://www.w3.org/WAI/WCAG22/Techniques/pdf/PDF1)

TPACでも話題に上がったルビを例に挙げれば、「1.3.1 情報及び関係性」の達成基準に対して、例えば「ルビと元の文字との関係性を示すためにruby要素を用いる（Using ruby elements to associate ruby text with base text)」というようなHTML向けのテクニックを新たに追加することが可能です。HTML以外の技術でも、実装方法が存在するなら同様に対処可能です。テクニックの新規追加は、随時誰でもAGWGに提案できます。
※逆に、よくある悪い例を失敗例（Failure = informative）として追加することも可能ではあります。

もともと、WCAG 2.ｘは、抽象度の高いWCAGを常にup to dateに保っていくために、各達成基準（normative）を満たすことができる具体的な方法（テクニック：Sufficient techniques）はinformativeにして随時追加・更新可能にしてある次第です。ただし、既存の文書化されたテクニックは、全ての方法を網羅できているわけではありません。

改めまして、明らかにアクセシビリティの問題として存在するが、(1) WCAG 2.2にはその問題に該当する達成基準がない、あるいは (2) 基準の数値等を日本語向けに変更すべきである、という事例があればご教示いただけましたら幸いです。
※年明けの仕事始め以降で大丈夫です。

**Comment by murata2makoto** [comment]
年内に概略だけお返事します。

ルビについてはfailure techniqueで対処しようという議論がTPACであったのはご存じの通りで、以下のissueがすでに上がっています。これの結論をWCAG JISではどう取り込むべきでしょうかね？

https://github.com/w3c/wcag/issues/4746

言語依存の振舞いについてregistryを使うと言う話が出たのもご存じだと思います。その後、Kevinはこの方向を断念しました。仕切り直しになったWCAG3 subgroupで話し合われる可能性はありますが見通しはたっていません。

詳細はまた来年にしましょう。良いお年を。

**Comment by MakotoUeki** [comment]
@murata2makoto 
> ルビについてはfailure techniqueで対処しようという議論がTPACであったのはご存じの通りで、以下のissueがすでに上がっています。これの結論をWCAG JISではどう取り込むべきでしょうかね？

ルビに限らず、Sufficient technique あるいは Common failure に追加があれば、WAICで順次翻訳していくと思いますが、TechniqueやFailureで解決できるレベルの事項に関しては、いずれにしてもInformativeなのでJIS改定には影響しない話だと考えます。

> 言語依存の振舞いについて

WCAGでの対応を優先するなら、現状ではおそらくWCAG3でということになるので、AGWGで年明けから次のフェーズに入っていくサブグループに参加されることをお奨めします。今回のJIS改定に関しては、当然間に合わないでしょう。

もしそれと並行して、JIS X 8341-3の改定版（Normative）でも何らかの対処が必要だというお考えであれば、まずは具体的な問題点の共有と対処法のご提案が急がれているという認識でおります。

ともかく、来年引き続き議論してまいりましょう。良いお年をお迎えください。

**Comment by momdo** [comment]
関連issue：
- #1902
- #1911

**Comment by bakera** [comment]
@murata2makoto 村田先生、コメントありがとうございます。JIS原案作成委員会の考えとして、以下に改めて状況をまとめさせていただきます。
 
- WAIC(原案作成団体)としては、既にヒアリングの場を設けさせていただくなど、村田先生のご意見については聴取を行っております。村田先生から頂いている内容は委員会、分科会でも共有し、いただいた内容への対応について議論しています
- ヒアリングの際にもお伝えした通り、根拠を伴う具体的な課題の提示、変更内容の提案をいただく必要があると考えております。いただいた課題・提案などをもって、委員会へのご参加や議事への追加を検討させていただきます
- この場は広く一般に公開しており、公開性についての懸念はあたらないと考えております。ただ、こちらに開示いただくことが難しいようであれば、直接メールにてご連絡頂けますようお願いいたします
 
（このgithubは技術的な内容や翻訳についての議論を意図していますので、原案作成委員会の運営に関してのご指摘・ご相談は何らかの別の場所を設けることも検討させてください）
